model/DataDifferencer.php

```diff
@@ -197,26 +199,29 @@ public function ChangedFields() {
 			if(!$this->fromRecord || $this->fromRecord->$field != $this->toRecord->$field) {
 				// Only show HTML diffs for fields which allow HTML values in the first place
 				$fieldObj = $this->toRecord->dbObject($field);
-				if($this->fromRecord) {
-					$fieldDiff = Diff::compareHTML(
-						$this->fromRecord->$field,
-						$this->toRecord->$field,
-						(!$fieldObj || $fieldObj->stat('escape_type') != 'xml')
-					);
-				} else {
-					if($fieldObj && $fieldObj->stat('escape_type') == 'xml') {
-						$fieldDiff = "<ins>" . $this->toRecord->$field . "</ins>";
+
+				if($fieldObj instanceof DBField) {
```

Comment:
The easiest way to do this is to move the `$this->db()` call in `dbObject` further up to the top of the method, and return null if `$helper` is empty (even if $record has a value).